app/assets/javascripts/editor.js

```diff
@@ -23,131 +26,122 @@ $E = {
+  }
+  image(src) {
     $E.wrap('\n![',']('+src+')\n')
-  },
-  h1: function() {
-    $E.wrap('#','')
-  },
-  h2: function() {
+  }
+  // these header formatting functions are not used anywhere, so commenting them out for now to pass codeclimate:
+
+  // h1() {
+  //   $E.wrap('#','')
+  // }
+  h2() {
     $E.wrap('##','')
-  },
-  h3: function() {
-    $E.wrap('###','')
-  },
-  h4: function() {
-    $E.wrap('####','')
-  },
-  h5: function() {
-    $E.wrap('#####','')
-  },
-  h6: function() {
-    $E.wrap('######','')
-  },
-  h7: function() {
-    $E.wrap('#######','')
-  },
+  }
+  // h3() {
+  //   $E.wrap('###','')
+  // }
+  // h4() {
+  //   $E.wrap('####','')
+  // }
+  // h5() {
+  //   $E.wrap('#####','')
+  // }
+  // h6() {
+  //   $E.wrap('######','')
+  // }
+  // h7() {
+  //   $E.wrap('#######','')
+  // }
   // this function is dedicated to Don Blair https://github.com/donblair
-  save: function() {
-    localStorage.setItem('plots:lastpost',$E.textarea.val())
-    localStorage.setItem('plots:lasttitle',$E.title.val())
-  },
-  recover: function() {
-    $E.textarea.val(localStorage.getItem('plots:lastpost'))
-    $E.title.val(localStorage.getItem('plots:lasttitle'))
-  },
-  apply_template: function(template) {
+  save() {
+    localStorage.setItem('plots:lastpost',$E.textarea.val());
+  }
+  recover() {
+    $E.textarea.val(localStorage.getItem('plots:lastpost'));
+  }
+  apply_template(template) {
     if($E.textarea.val() == ""){
       $E.textarea.val($E.templates[template])
     }else if(($E.textarea.val() == $E.templates['event']) || ($E.textarea.val() == $E.templates['default']) || ($E.textarea.val() == $E.templates['support'])){
         $E.textarea.val($E.templates[template])
     }else{
       $E.textarea.val($E.textarea.val()+'\n\n'+$E.templates[template])
     }
-  },
-  templates: {
-    'blog': "## The beginning\n\n## What we did\n\n## Why it matters\n\n## How can you help",
-    'default': "## What I want to do\n\n## My attempt and results\n\n## Questions and next steps\n\n## Why I'm interested",
-    'support': "## Details about the problem\n\n## A photo or screenshot of the setup",
-    'event': "## Event details\n\nWhen, where, what\n\n## Background\n\nWho, why",
-    'question': "## What I want to do or know\n\n## Background story"
-  },
-  previewing: false,
-  previewed: false,
-  generate_preview: function(id,text) {
+  }
+  generate_preview(id,text) {
     $('#'+id)[0].innerHTML = marked(text)
-  },
-  toggle_preview: function() {
+  }
+  toggle_preview() {
     let previewBtn;
     let dropzone;
     // if the element is part of a multi-comment page,
     // ensure to grab the current element and not the other comment element.
-    previewBtn = $(this.textarea.context).find('.preview-btn');
-    dropzone = $(this.textarea.context).find('.dropzone');
+    previewBtn = $("#toggle-preview-button-" + this.commentFormID);
+    dropzone = $("#dropzone-large-" + this.commentFormID);
```

Comment:
Rewrote these assignments so it sets on a specific ID, instead of searching through the DOM.